workflows/targeted.nf

```diff
@@ -0,0 +1,693 @@
+
+    //
+    // SUBWORKFLOW: Run read alignment to generate BAMs
+    //
+    // channel: [ meta, [bam, ...], [bai, ...] ]
+    ch_align_dna_tumor_out = Channel.empty()
+    ch_align_dna_normal_out = Channel.empty()
+    ch_align_rna_tumor_out = Channel.empty()
+    if (run_config.stages.alignment) {
+
+        READ_ALIGNMENT_DNA(
+            ch_inputs,
+            ref_data.genome_fasta,
+            ref_data.genome_bwamem2_index,
+            params.max_fastq_records,
+        )
+
+        READ_ALIGNMENT_RNA(
+            ch_inputs,
+            ref_data.genome_star_index,
+        )
+
+        ch_versions = ch_versions.mix(
+            READ_ALIGNMENT_DNA.out.versions,
+            READ_ALIGNMENT_RNA.out.versions,
+        )
+
+        ch_align_dna_tumor_out = ch_align_dna_tumor_out.mix(READ_ALIGNMENT_DNA.out.dna_tumor)
+        ch_align_dna_normal_out = ch_align_dna_normal_out.mix(READ_ALIGNMENT_DNA.out.dna_normal)
+        ch_align_rna_tumor_out = ch_align_rna_tumor_out.mix(READ_ALIGNMENT_RNA.out.rna_tumor)
+
+    } else {
+
+        ch_align_dna_tumor_out = ch_inputs.map { meta -> [meta, [], []] }
+        ch_align_dna_normal_out = ch_inputs.map { meta -> [meta, [], []] }
+        ch_align_rna_tumor_out = ch_inputs.map { meta -> [meta, [], []] }
+
+    }
+
+    //
+    // SUBWORKFLOW: Run MarkDups for DNA BAMs
+    //
+    // channel: [ meta, bam, bai ]
+    ch_process_dna_tumor_out = Channel.empty()
+    ch_process_dna_normal_out = Channel.empty()
+    if (run_config.stages.markdups) {
+
+        // NOTE(SW/MC): hardcoded for initial testing purposes
+        has_umis = run_config.panel.equalsIgnoreCase('tso500')
+
+        READ_PROCESSING(
+            ch_inputs,
+            ch_align_dna_tumor_out,
+            ch_align_dna_normal_out,
+            ref_data.genome_fasta,
+            ref_data.genome_version,
+            ref_data.genome_fai,
+            ref_data.genome_dict,
+            hmf_data.unmap_regions,
+            has_umis,
+        )
+
+        ch_versions = ch_versions.mix(READ_PROCESSING.out.versions)
+
+        ch_process_dna_tumor_out = ch_process_dna_tumor_out.mix(READ_PROCESSING.out.dna_tumor)
+        ch_process_dna_normal_out = ch_process_dna_normal_out.mix(READ_PROCESSING.out.dna_normal)
+
+    } else {
+
+        ch_process_dna_tumor_out = ch_inputs.map { meta -> [meta, [], []] }
+        ch_process_dna_normal_out = ch_inputs.map { meta -> [meta, [], []] }
+
+    }
+
+    //
+    // MODULE: Run Isofox to analyse RNA data
+    //
+    // channel: [ meta, isofox_dir ]
+    ch_isofox_out = Channel.empty()
+    if (run_config.stages.isofox) {
+
+        isofox_counts = params.isofox_counts ? file(params.isofox_counts) : panel_data.isofox_counts
+        isofox_gc_ratios = params.isofox_gc_ratios ? file(params.isofox_gc_ratios) : panel_data.isofox_gc_ratios
+        isofox_read_length = params.isofox_read_length !== null ? params.isofox_read_length : Constants.DEFAULT_ISOFOX_READ_LENGTH_TARGETED
```

Comment:
Does this need the "!== null"?

Reply:
This is needed to guarantee safety since isofox_read_length (but not the other two variables/params above) is initially set to null but then given a default value conditioned on the analysis mode (targeted: 93; wgts: 151). Both the isofox_counts and isofox_gc_ratios variables are directly assigned defaults rather than null value so do not need this additional check.